test/extended/apibinding/apibinding_utils.go

```diff
@@ -81,10 +81,12 @@ func (apb *APIBinding) Create(k *exutil.CLI) {
 func (apb *APIBinding) CreateAsExpectedResult(k *exutil.CLI, successFlag bool, containsMsg string) {
 	outputFile, err := exutil.StructMarshalOutputToFile(apb, apb.Metadata.Name)
 	o.Expect(err).NotTo(o.HaveOccurred())
-	msg := exutil.ApplyResourceFromTemplate(k, outputFile)
+	msg, applyErr := exutil.ApplyResourceFromTemplate(k, outputFile)
 	if successFlag {
+		o.Expect(applyErr).ShouldNot(o.HaveOccurred())
 		o.Expect(msg).Should(o.ContainSubstring(containsMsg))
 	} else {
+		o.Expect(applyErr).Should(o.HaveOccurred())
```

Comment:
Good catch that @zimo-xiao's `resourceFromTemplate` will make your case fail here within `else`, since that always assumes success. Seems his updates only affect your CreateAsExpectedResult? If so, how about only replacing the call of ApplyResourceFromTemplate in line 84 to the three lines in your https://github.com/kcp-dev/kcp-tests/pull/8/files#diff-e21320d17fdce22cad99f368c20f8aca8b28bac96f890d69e95fda9be0ae7f93R906-R908 given three lines are not too many? By this way, other "xxxFromTemplatexxx" functions need not be followed up with an `o.Expect(err).NotTo(o.HaveOccurred())`?

Reply:
BTW @Phaow your PRs try to test all cases to pass which helps find failure, this is good action :+1: . @zxiao-redhat , if possible and if it does not take much time to test through all cases given currently our case number is not too many, could you also do same when testing a PR instead of only the exact case of a PR? E.g. if https://github.com/kcp-dev/kcp-tests/pull/19 does this, then it can catch the failure. Of course, it it takes too much time to test through all cases, then ignore this. Just a suggestion :) . CC @kasturinarra .

Reply:
@xingxingxia Thanks! IMO, the underlayer framework utils not process the error with assert, return it directly is better, as it'll use for all areas/scenarios. We could expand it with a new method(in sub area folder) just add the assert or I could use a new function without this assert impel my methods. Both ok for me, we'd better make a rule for this to make our codes the same style.

Reply:
@xingxingxia sure, thanks for pointing out. Will fully test before merge. @Phaow this util was migrated from open shift ginkgo. I already saw issues relating to errors not returning. This is a good fix.